ProcessLib/GroundwaterFlow/GroundwaterFlowFEM.h

```diff
@@ -64,21 +77,31 @@ class LocalAssemblerData : public ProcessLib::LocalAssemblerInterface<GlobalMatr
         assert(local_matrix_size == ShapeFunction::NPOINTS * NUM_NODAL_DOF);
     }
 
-    void assemble(double const /*t*/, std::vector<double> const& /*local_x*/) override
+    void assemble(double const /*t*/, std::vector<double> const& local_x) override
     {
         _localA.setZero();
         _localRhs.setZero();
 
         IntegrationMethod integration_method(_integration_order);
         unsigned const n_integration_points = integration_method.getNPoints();
 
-        for (std::size_t ip(0); ip < n_integration_points; ip++) {
+        for (std::size_t ip(0); ip < n_integration_points; ip++)
+        {
             auto const& sm = _shape_matrices[ip];
             auto const& wp = integration_method.getWeightedPoint(ip);
-
             auto const k = _process_data.hydraulic_conductivity(_element);
+
             _localA.noalias() += sm.dNdx.transpose() * k * sm.dNdx *
                                  sm.detJ * wp.getWeight();
+
+            // Darcy velocity only computed for output.
+            auto const darcy_velocity = (k * sm.dNdx *
```

Comment:
> If we deal with non-linear problems, we only need to compute the velocity after a linearization step is converged. 

That's an interesting fact, thanks. I didn't know it.

Regarding the GWFlow files in this PR: Any changes made are primarily for illustrative purposes. Since the velocity is not needed in GWFlow it probably should not be computed in production code at all. Besides, the current way of computing it produces completely meaningless results, since the velocity is computed before solving the equation system.

I think it doesn't make much sense to extend GWFlow by a `postTimestep()` hook that computes the velocity which is needed only for output. I suggest that I keep the GWFlow changes during the review, but will remove it before merging. This way you'll see how secondary variables can be used and the current code does not get cluttered in the end.